tests/foreman/longrun/test_oscap.py

```diff
@@ -493,3 +495,130 @@ def test_positive_reporting_emails_of_oscap_reports():
 
     :CaseLevel: System
     """
+
+
+@pytest.mark.parametrize('distro', ['rhel8'])
+def test_positive_oscap_run_via_local_files(
+    module_org, default_proxy, content_view, lifecycle_env, distro, module_target_sat
+):
+    """End-to-End Oscap run via local files deployed with ansible
+
+    :id: 0dde5893-540c-4e03-a206-55fccdb2b9ca
+
+    :parametrized: yes
+
+    :customerscenario: true
+
+    :setup: scap content, scap policy , Remote execution
+
+    :steps:
+
+        1. Create a valid scap content
+        2. Import Ansible role theforeman.foreman_scap_client
+        3. Create a scap policy with ansible as deploy option
+        4. Associate the policy with a hostgroup
+        5. Run the Ansible job and then trigger the Oscap job.
+        6. Oscap must Utilize the local files for the client scan.
+
+    :expectedresults: Oscap run should happen using the --localfile argument.
+
+    :BZ: 2081777,2211952
+
+    :CaseImportance: Critical
+    """
+    SELECTED_ROLE = 'theforeman.foreman_scap_client'
+    file_name = 'security-data-oval-com.redhat.rhsa-RHEL8.xml.bz2'
+    download_url = 'https://www.redhat.com/security/data/oval/com.redhat.rhsa-RHEL8.xml.bz2'
+    profile = OSCAP_PROFILE['ospp8']
+    content = OSCAP_DEFAULT_CONTENT[f'{distro}_content']
+    hgrp_name = gen_string('alpha')
+    policy_name = gen_string('alpha')
+
+    module_target_sat.cli_factory.make_hostgroup(
+        {
+            'content-source-id': default_proxy,
+            'name': hgrp_name,
+            'organizations': module_org.name,
+        }
+    )
+    # Creates oscap_policy.
+    scap_id, scap_profile_id = fetch_scap_and_profile_id(content, profile)
+    with Broker(
+        nick=distro,
+        host_class=ContentHost,
+        target_cores=OSCAP_TARGET_CORES,
+        target_memory=OSCAP_TARGET_MEMORY,
```

Comment:
Missing change from your previous PR, will commit this suggestion before running PRT
```suggestion
        deploy_flavor=settings.flavors.default,
```